src/Microsoft.Management.Configuration/Microsoft.Management.Configuration.idl

```diff
@@ -0,0 +1,605 @@
+    runtimeclass ApplySettingsResult
+    {
+        ApplySettingsResult();
+
+        // Indicates whether a reboot is required after the settings were applied.
+        Boolean RebootRequired;
+
+        // The result of applying the configuration unit.
+        ConfigurationUnitResultInformation ResultInformation{ get; };
+    }
+
+    // Informs the caller of the result of running a Test.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    enum ConfigurationTestResult
+    {
+        // The result is unknown.
+        Unknown,
+        // The system is in the state described by the configuration.
+        Positive,
+        // The system is not in the state described by the configuration.
+        Negative,
+        // Running the test failed.
+        Failed,
+        // The test was not run because it was not applicable.
+        NotRun,
+    };
+
+    // The result of testing the settings with an IConfigurationUnitProcessor.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    runtimeclass TestSettingsResult
+    {
+        TestSettingsResult();
+
+        // The result (if any) of running Test on the configuration unit.
+        ConfigurationTestResult TestResult;
+
+        // The result of testing the configuration unit.
+        // This is not the response for the test, but rather contains information about the actual attempt to run the test.
+        ConfigurationUnitResultInformation ResultInformation{ get; };
+    }
+
+    // The result of getting the settings with an IConfigurationUnitProcessor.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    runtimeclass GetSettingsResult
+    {
+        GetSettingsResult();
+
+        // The current state of the system for the configuration unit.
+        Windows.Foundation.Collections.ValueSet Settings;
+
+        // The result of getting the configuration unit settings.
+        // This is not the response for the retrieval, but rather contains information about the actual attempt to retrieve the settings.
+        ConfigurationUnitResultInformation ResultInformation{ get; };
+    }
+
+    // Provides access to a specific configuration unit within the runtime.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    interface IConfigurationUnitProcessor
+    {
+        // The configuration unit that the processor was created for.
+        ConfigurationUnit Unit{ get; };
+
+        // The directives overlay that the processor was created with.
+        Windows.Foundation.Collections.IMapView<String, Object> DirectivesOverlay{ get; };
+
+        // Determines if the system is already in the state described by the configuration unit.
+        TestSettingsResult TestSettings();
+
+        // Gets the current system state for the configuration unit.
+        GetSettingsResult GetSettings();
+
+        // Applies the state described in the configuration unit.
+        ApplySettingsResult ApplySettings();

```

Comment:
Here we use Test/Get/Apply, but elsewhere it was Assert/Inform/Apply

Reply:
I wanted to separate the semantics, but maybe it is also confusing...